.github/workflows/continuous-integration.yaml

```diff
@@ -451,6 +451,136 @@ jobs:
+      steps:
+        - name: Configure AWS credentials
+          uses: aws-actions/configure-aws-credentials@ececac1a45f3b08a01d2dd070d28d111c5fe6722
+          with:
+            aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+            aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+            aws-region: ${{ secrets.AWS_REGION }}
+              
+        - name: Start SLAB GPU runner
+          id: start-slab-runner
+          if: ${{ !cancelled() }}
+          uses: zama-ai/slab-github-runner@79939325c3c429837c10d6041e4fd8589d328bac
+          with:
+              mode: start
+              github-token: ${{ secrets.SLAB_ACTION_TOKEN }}
+              slab-url: ${{ secrets.SLAB_BASE_URL }}
+              job-secret: ${{ secrets.JOB_SECRET }}
+              backend: aws
+              profile: gpu_ciprofile 
+
+  # GPU-availability:
+  #   name: GPU Availability
+  #   runs-on: ${{ needs.start-runner-gpu-linux.outputs.label }}
+  #   needs: start-runner-gpu-linux
+    
+  #   steps:
+  #     - name: Check GPU availability
+  #       run: |
+  #         echo "Checking for GPU..."
+  #         nvidia-smi || echo "nvidia-smi failed"
+
+  GPU-tests:
+    name: Run GPU tests (Python ${{ matrix.python_version }})
+    runs-on: ${{ needs.start-runner-gpu-linux.outputs.label }}
+    needs: [start-runner-gpu-linux]
+    permissions:
+      contents: read
+    strategy:
+      matrix:
+        python_version: [3.8]
+
+    steps:
+      - name: Set Home
+        run: |
+          echo "Set HOME=$(pwd)"
+          echo "HOME=$(pwd)" >> $GITHUB_ENV
+
+      - name: Disable LFS download by default
+        run: |
+
+          apt-get update
+          apt-get install git-lfs
+
+          git lfs install --skip-smudge
+
+      - name: Checkout code
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683
+
+      - name: Pull LFS files
+        run: |
+          git lfs pull --include "tests/data/**, src/concrete/ml/pandas/_client_server_files/**" --exclude  ""
+
+      - name: Set up Python ${{ matrix.python_version }}
+        uses: actions/setup-python@42375524e23c412d93fb67b49958b491fce71c38
+        id: setup-python
+        with:
+          python-version: ${{ matrix.python_version }}
+
+      - name: Install Deps
+        id: install-deps
+        run: |
+          echo "Using these tools: $(which python3), $(which pip3)"      
+          ./script/make_utils/setup_os_deps.sh
+          make setup_env
+
+      - name: Install GPU concrete-python
+        id: install-gpu-concrete-python
+        run: |
+          # Show CP version
+          poetry run pip show concrete-python || echo "concrete-python not installed"
+          poetry run pip uninstall -y concrete-python
+
+          
+          # poetry run pip install --extra-index-url https://pypi.zama.ai/gpu concrete-python
+          poetry run pip install --extra-index-url https://pypi.zama.ai/gpu concrete-python==2.10.0
```

Comment:
we need to get the version that is currently install and install == that version

something like:

```
CONCRETE_WITH_VERSION= $(pip freeze | grep concrete-python)
poetry run pip uninstall -y concrete-python
poetry run pip install --extra-index-url https://pypi.zama.ai/gpu $CONCRETE_WITH_VERSION
```